File: lua/database/utils/table.lua
--
--  Copyright (C) 2024 Command_maker
--
-- This program is free software: you can redistribute it andor modify
-- it under the terms of the GNU General Public License as published by
-- the Free Software Foundation, either version 3 of the License, or
-- (at your option) any later version.
-- This program is distributed in the hope that it will be useful,
-- but WITHOUT ANY WARRANTY; without even the implied warranty of
-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-- GNU General Public License for more details.
--
-- You should have received a copy of the GNU General Public License
-- along with this program.  If not, see <http:/www.gnu.org/licenses/>.

local tableLib = {}

function tableLib.contains(t, val)
    for _, v in ipairs(t) do
        if val == v then
            return true
        end
    end

    return false
end

function tableLib.unique(t)
    local out = {}

    for _, val in ipairs(t) do
        if not tableLib.contains(out, val) then
            table.insert(out, val)
        end
    end

    return out
end

return tableLib

File: python/models/table.py
#
#  copyright (c) 2024 command_maker
#
# this program is free software: you can redistribute it andor modify
# it under the terms of the gnu general public license as published by
# the free software foundation, either version 3 of the license, or
# (at your option) any later version.
# this program is distributed in the hope that it will be useful,
# but without any warranty; without even the implied warranty of
# merchantability or fitness for a particular purpose.  see the
# gnu general public license for more details.
#
# you should have received a copy of the gnu general public license
# along with this program.  if not, see <http:/www.gnu.org/licenses/>.


class SQLTable:

    columns: list[str]
    table_name: str
    extra_args: list[str] = [] 

    def __init__(self, table_obj: dict) -> None:
        self.table_name = table_obj['table_name']
        self.columns = table_obj['columns']
        self.extra_args = table_obj['extra_args'] if table_obj.__contains__('extra_args') else []
